proto/process/agent.proto

```diff
@@ -1060,6 +1078,92 @@ message StatefulSet {
 	repeated string tags = 5;
 }
 
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2576
+message PersistentVolume {
+	Metadata metadata = 1;
+  PersistentVolumeSpec spec = 2;
+  PersistentVolumeStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
+}
+
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2889
+message PersistentVolumeSpec {
+  // Key/Value of a resource type and a corresponding quantity as an int64.
+  map<string, int64> capacity = 1;
+  // in kubernetes these are actually fields in a struct where only one can be set.
+  // For now we only set it to a string with the name.
+  string persistentVolumeSource = 2;
+  repeated string accessModes = 3;
+  ObjectReference claimRef = 4;
+  string persistentVolumeReclaimPolicy = 5;
+  string storageClassName = 6;
+	repeated string mountOptions = 7;
+  string volumeMode = 8;
+  repeated NodeSelectorTerm nodeSelectorTerms = 9;
+}
+
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2940
+message PersistentVolumeStatus {
+	string phase = 1;
+	string message = 2;
+	string reason = 3;
+}
+
+message NodeSelectorTerm{
+  repeated LabelSelectorRequirement matchExpressions = 1;
+  repeated LabelSelectorRequirement matchFields = 2;
+}
+
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2597
+message PersistentVolumeClaim {
+  Metadata metadata = 1;
+  PersistentVolumeClaimSpec spec = 2;
+  PersistentVolumeClaimStatus status = 3;
+  bytes yaml = 4;
+  repeated string tags = 5;
+}
+
+
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2715
+message PersistentVolumeClaimStatus {
+  string phase = 1;
+	repeated string accessModes = 2;
+  map<string, int64> capacity = 3;
+  repeated PersistentVolumeClaimCondition conditions = 4;
+}
+
+// reference https://github.com/kubernetes/api/blob/4bdcbc60ffeee2c04231f11ff114314c1a93ce65/core/v1/generated.proto#L2654
+message PersistentVolumeClaimSpec {
```

Comment:
might be useful to try and collect the `finalizers` see https://kubernetes.io/docs/concepts/storage/persistent-volumes/#storage-object-in-use-protection

Reply:
The `finalizers` are in the `ObjectMetadata` struct. Do you suggest to either:
- update our Metadata collection to include finaliziers?
- collect the `finalizers` as its own field for PVC  

I prefer the former as it might be useful for other resources. 🤔

Reply:
added here: https://github.com/DataDog/agent-payload/pull/115/commits/21c8ddcf51ec9029eda96c0ef60db5891bf3cfc1